apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.TableCrud.cs

```diff
@@ -213,16 +230,31 @@ private ListRequest CreateListTablesRequest(DatasetReference datasetReference, L
             return request;
         }
 
-        private InsertRequest CreateInsertTableRequest(TableReference tableReference, TableSchema schema, CreateTableOptions options)
+        private InsertRequest CreateInsertTableRequest(TableReference tableReference, Table resource, CreateTableOptions options)
         {
-            GaxPreconditions.CheckNotNull(tableReference, nameof(tableReference));
+            CheckResourceForCreation(tableReference, resource);
+
+            resource.TableReference ??= tableReference;
```

Comment:
It's worth noting this modifies the resource passed in by the user. We *may* want to document that. I don't think it's massively critical though.

Reply:
Done.